src/resdk/tables/variant.py

```diff
@@ -79,90 +80,187 @@ class VariantTables(BaseTables):
+            # Assign geneset from the Genialis Platform
+            self.geneset = self._get_obj_geneset(self._data[0])
+        else:
+            self.geneset = geneset
 
-        self.mutations = mutations
-        if mutations is not None:
-            if not isinstance(mutations, list):
-                raise ValueError("Mutations must be a list of strings")
+    @property
+    def geneset(self):
+        """Get geneset."""
+        return self._geneset
+
+    @geneset.setter
+    def geneset(self, value):
+        """Set geneset."""
+        # Geneset can be set only once, prevent modifications
+        if self._geneset is not None:
+            raise ValueError("It is not allowed to change geneset value.")
+
+        if value is None:
+            return
+
+        # If id / slug of a geneset is given, get it from the Resolwe server
+        if isinstance(value, (int, str)):
+            gs = self.resolwe.geneset.get(value)
+            value = gs.genes
+        elif isinstance(value, Geneset):
+            value = value.genes
+
+        if isinstance(value, (list, set, tuple, pd.Series)):
+            self._geneset = set(value)
         else:
-            # Check for heterogeneous mutations only if mutations is not specified
-            self.check_heterogeneous_mutations()
+            raise ValueError(f'Unsupported type of "geneset" input: {value}.')
+
+    @property
+    @lru_cache()
+    def variants(self) -> pd.DataFrame:
+        """Get variants table.
+
+        There are 4 possible values:
+
+            - 0 - wild-type, no variant
+            - 1 - heterozygous mutation
+            - 2 - homozygous mutation
+            - NaN - QC filters are failing - mutation status is unreliable
+
+        """
+        df = self._load_fetch(self.VARIANTS)
+
+        # Variants that are not reported (NaN) were not detected:
+        # they are wild type.
+        df = df.fillna(0)
+
+        if self.filtering:
+            # Keep values in case .filter == PASS or .variants == 0
+            df = df.where((self.filter == "PASS") | (df == 0), other=np.nan)
+
+        return df
+
+    @property
+    @lru_cache()
+    def depth(self) -> pd.DataFrame:
+        """Get depth table."""
+        return self._load_fetch(self.DEPTH)
+
+    @property
+    @lru_cache()
+    def depth_a(self) -> pd.DataFrame:
+        """Get depth table for adenine."""
+        return self._load_fetch(self.DEPTH_A)
+
+    @property
+    @lru_cache()
+    def depth_c(self) -> pd.DataFrame:
+        """Get depth table for cytosine."""
+        return self._load_fetch(self.DEPTH_C)
 
-    def check_heterogeneous_mutations(self):
-        """Ensure variants are selected for the same mutations in all samples."""
-        mutations = {str(d.input["mutations"]) for d in self._data}
+    @property
+    @lru_cache()
+    def depth_g(self) -> pd.DataFrame:
+        """Get depth table for guanine."""
+        return self._load_fetch(self.DEPTH_G)
+
```

Comment:
Just curious: is it easier to user five methods (depth[acgt]?) or just one (depth) with optional argument specifying the amino acid?

Reply:
I see that the methods were already  in use before, so please ignore the previous question.